index.js

```diff
@@ -1,255 +1,270 @@
-    event.Records[0].s3.object.key = decodeURIComponent(event.Records[0].s3.object.key.replace(/\+/g, ' '));
-    return {
-        event: {
-            version: event.Records[0].eventVersion
-            , source: event.Records[0].eventSource
-            , time: event.Records[0].eventTime
-            , name: event.Records[0].eventName
-        }
-        , s3: {
-            region: event.Records[0].awsRegion
-            , user: event.Records[0].userIdentity.principalId
-            , schema: event.Records[0].s3.s3SchemaVersion
-            , configuration: event.Records[0].s3.configurationId
-        }
-        , bucket: {
-            name: event.Records[0].s3.bucket.name
-            , owner: event.Records[0].s3.bucket.ownerIdentity.principalId
-            , arn: event.Records[0].s3.bucket.arn
-        }, object: event.Records[0].s3.object
-    };
-};
+  event.Records[0].s3.object.key = decodeURIComponent(event.Records[0].s3.object.key.replace(/\+/g, ' '))
+  return {
+    event: {
+      version: event.Records[0].eventVersion
+    , source: event.Records[0].eventSource
+    , time: event.Records[0].eventTime
+    , name: event.Records[0].eventName
+    }
+  , s3: {
+      region: event.Records[0].awsRegion
+    , user: event.Records[0].userIdentity.principalId
+    , schema: event.Records[0].s3.s3SchemaVersion
+    , configuration: event.Records[0].s3.configurationId
+    }
+  , bucket: {
+      name: event.Records[0].s3.bucket.name
+    , owner: event.Records[0].s3.bucket.ownerIdentity.principalId
+    , arn: event.Records[0].s3.bucket.arn
+    }, object: event.Records[0].s3.object
+  }
+}
 
 // Process Event Data After Parsing
 const processEvent = (eventData) => {
-    return {
-        file: `${eventData.bucket.name}/${eventData.object.key}`
-        , timestamp: (new Date(eventData.event.time)).getTime()
-        , meta: {
-            bucket: eventData.bucket
-            , object: eventData.object
-            , region: eventData.s3.region
-            , source: eventData.event.source
-            , user: eventData.event.user
-        }
-    };
-};
+  return {
+    file: `${eventData.bucket.name}/${eventData.object.key}`
+  , timestamp: (new Date(eventData.event.time)).getTime()
+  , meta: {
+      bucket: eventData.bucket
+    , object: eventData.object
+    , region: eventData.s3.region
+    , source: eventData.event.source
+    , user: eventData.event.user
+    }
+  }
+}
 
 // Prepare the Messages and Options
 const prepareEvent = (eventData) => {
-    return Object.assign({}, eventData, {
-        line: formatMessage(eventData)
-    });
-};
+  return {...eventData, line: formatMessage(eventData)}
+}
+
+// Extra the Main, Meta, and Time Data
+const extractData = (data) => {
+  const {Records, logFiles, ...meta} = data
+  const timestamp = meta && meta.eventTime ? (new Date(meta.eventTime)).getTime() : undefined
+  data = Records || logFiles || data
+  return Array.isArray(data) ? data.reduce((lines, entry) => {
```

Comment:
This feels more like a `map` operation (like it was before) than a `reduce` one.

Reply:
Done!